### PR TITLE
[RNMobile] Add mobile equivalent UnitControl 

### PR DESCRIPTION
**Pull request by lukewalczak**
## Description

That PR is an introduction to implementing different units on mobile platform. It includes:

* new mobile hook called `useConvertUnitToMobile` which translates each unit to RN and returns the number which can be then applied to `style`
* new component `UnitControl` - pressing the unit opens the action sheet:

ios | android
--- | ---
new-design | new-design-android

* extracted input to the separate component called `RangeTextInput` which gives option to easier define the value:

ios | android
--- | ---
Screenshot 2020-10-22 at 14 53 30 | Screenshot 2020-10-22 at 14 54 37

* new changes are applied to following blocks:
1. `Column` (_user can change the value but there is no visual effect_)
2. `Cover`

 

## How has this been tested?

Cover the following tests from sanity testing:

- [x] Cover - Change Minimum height in pixels - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/cover.md#tc002)

- [x] Columns - Max limit of Column number - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/columns.md#tc005)
- [x] Columns - Min limit of Column number - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/columns.md#tc006)

- [x] Spacer block - Spacer is rendered without crash - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc001)
- [x] Spacer block - Settings: Control spacer height using the slider - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc002)
- [x] Spacer block - Settings: Control spacer height using the text input - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc003)
- [x] Spacer block - Settings: Available height range is correct - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc004)
- [x] Spacer block - Settings: Height range extends if Spacer comes from the web is higher than 500px - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/spacer.md#tc005)

What's more:

1. Open a mobile app (wp-ios / wp-android)
2. Add `Cover` block
3. Choose the color from placeholder
4. Open settings
5. Change the height to 290 using stepper
6. Change the height to 300 using stepper
**Expect that Cover height is changing**
7. Change the unit to other unit eg `vw`
**Expect that Cover height is changing**
8. Open html mode
**Expect that chosen unit is set in HTML**

## Screenshots 

_Available in description section_

## Types of changes

New feature:
* two new components
* new mobile hook

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by lukewalczak** [comment]
**UPDATE**

Added previews on the left side of the cell title:

ios | android
--- | ---
Screenshot 2020-10-27 at 09 49 09 | Screenshot 2020-10-27 at 09 48 41

**Comment by lukewalczak** [comment]
**UPDATES**

1. There is no more `Menu` component on Android -> we switch back to `BottomSheet`:

light mode | dark mode
--- | ---
Screenshot 2020-11-02 at 13 58 02 | Screenshot 2020-11-02 at 13 19 00

2. Android stepper icons size are increased from `18` to `24` and there are also spacing refinements:
Screenshot 2020-11-02 at 13 18 39

3. Width values aren't cut off

File: packages/block-library/src/columns/edit.native.js [hunk 1]
```diff
@@ -148,32 +158,50 @@ function ColumnsEditContainer( {
 		return null;
 	};
 
-	const getColumnsSliders = () => {
-		const columnWidths = innerColumns.map(
-			( innerColumn ) =>
-				parseFloat( innerColumn.attributes.width ) || 100 / columnCount
+	const onChangeWidth = ( nextWidth, valueUnit, columnId ) => {
+		nextWidth = 0 > parseFloat( nextWidth ) ? '0' : nextWidth;
```
Comment (dratwas) [inline packages/block-library/src/columns/edit.native.js:162]: It looks the same as https://github.com/WordPress/gutenberg/pull/26161/files#diff-0a31b6b4b1dea0b86812a4333da8edd1ff843026c6d59fa14453ebd6ff5a4130R64

Have you considered to move it to some util function that takes 2 args (nextWidth, valueUnit) and returns string  `${ nextWidth }${ valueUnit }`?

File: packages/block-library/src/cover/shared.js [hunk 2]
```diff
@@ -23,12 +25,34 @@ export function backgroundImageStyles( url ) {
...
 
+const isWeb = Platform.OS === 'web';
+
 export const CSS_UNITS = [
-	{ value: 'px', label: 'px', default: 430 },
-	{ value: 'em', label: 'em', default: 20 },
-	{ value: 'rem', label: 'rem', default: 20 },
-	{ value: 'vw', label: 'vw', default: 20 },
-	{ value: 'vh', label: 'vh', default: 50 },
+	{
```
Comment (dratwas) [inline packages/block-library/src/cover/shared.js:31]: Did you remove the defaults intentionally?

File: packages/components/src/mobile/bottom-sheet/range-cell.native.js [hunk 3]
```diff
@@ -5,10 +5,7 @@ import {
 	Platform,
```
Comment (dratwas) [inline packages/components/src/mobile/bottom-sheet/range-cell.native.js:5]: I'm wondering if this refactor is a necessary part of this PR

Comment (dratwas) [reply packages/components/src/mobile/bottom-sheet/range-cell.native.js:5]: Nvm, i see that it is needed :)

File: packages/components/src/unit-control/index.native.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	separatorType,
+	units = CSS_UNITS,
+	unit,
+	getStylesFromColorScheme,
+	decimalNum,
+	...props
+} ) {
+	const pickerRef = useRef();
+
+	function onPickerSelect( selectedOption ) {
```
Comment (dratwas) [inline packages/components/src/unit-control/index.native.js:37]: Do you need that function? Why you don't use the `onUnitChange` directly?

File: packages/block-library/src/cover/shared.js (hunk 2 above)
Comment (lukewalczak) [reply packages/block-library/src/cover/shared.js:31]: Totally no, good catch!

File: packages/components/src/unit-control/index.native.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	const renderUnitPicker = () => {
+		return (
+			<View style={ styles.unitMenu }>
+				{ renderUnitButton() }
+				<Picker
+					ref={ pickerRef }
+					options={ units }
+					onChange={ onPickerSelect }
+					hideCancelButton
+					leftAlign={ true }
```
Comment (dratwas) [inline packages/components/src/unit-control/index.native.js:76]:
```suggestion
					leftAlign
```

File: packages/components/src/unit-control/index.native.js (hunk 4 above)
Comment (lukewalczak) [reply packages/components/src/unit-control/index.native.js:37]: It looks like the leftover after switching back from `Menu` to  `BottomSheet`, will move it directly to the mentioned function.

**Comment by iamthomasbishop** [comment]
@lukewalczak Those updates look solid, nicely done!

**Review by dratwas — APPROVED** [review]
I tested it on iOS and Android and everything works as expected. Great job Luke! 🎉

**Review by geriux — APPROVED** [review]
LGTM! Tested it on both iOS and Android, really nice work Luke! 👏 🚀

**Pull request merged**